CMakeLists.txt

```diff
@@ -511,7 +511,7 @@ if(AVIF_BUILD_APPS OR (AVIF_BUILD_TESTS AND AVIF_ENABLE_GTEST))
         avif_apps STATIC apps/shared/avifjpeg.c apps/shared/iccjpeg.c apps/shared/avifpng.c apps/shared/avifutil.c
                          apps/shared/y4m.c
     )
-    target_link_libraries(avif_apps avif ${AVIF_PLATFORM_LIBRARIES} ${PNG_LIBRARY} ${ZLIB_LIBRARY} ${JPEG_LIBRARY})
+    target_link_libraries(avif_apps avif_internal ${AVIF_PLATFORM_LIBRARIES} ${PNG_LIBRARY} ${ZLIB_LIBRARY} ${JPEG_LIBRARY})
```

Comment:
The code in apps/shared/ is considered as a client of the libavif library and should call the public functions of libavif. So `avif_apps` should not depend on the `avif_internal` library. Another problem with this change is that if `BUILD_SHARED_LIBS` is on, then the `avifenc` and `avifdec` apps will be linked with two copies of libavif. The first copy is the shared `avif` library, and the second copy is the static `avif_internal` library (via `apps_shared`).

I haven't fully analyzed the code in this pull request, but a good strategy is to move as many of the Exif functions declared in internal.h as we can to apps/shared/ (say a new apps/shared/avifexif.c file), and export some of the remaining Exif functions as public API functions (and declare them in avif.h). This may require refactoring the Exif functions.

The `avif_internal` library should only be used by unit tests of internal functions.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1199.